nucypher/blockchain/eth/utils.py

```diff
@@ -85,18 +85,19 @@ def estimate_block_number_for_period(period: int, seconds_per_period: int,  late
 def etherscan_url(item, network: str, is_token=False) -> str:
     if network is None or network is UNKNOWN_DEVELOPMENT_CHAIN_ID:
         raise ValueError("A network must be provided")
-    elif network == 'mainnet':
+
+    network = network.lower()
```

Comment:
Seems like a red flag that we have to do it for something coming from inside our own code and not from the user. `blockchain/eth/clients.py` already has a `# TODO: Consider creating a ChainInventory class and/or moving this to a separate module`, which would certainly help.

Reply:
You're totally right, let's do this :)

Reply:
I included an additional commit that checks wrt to the chain names dictionary. We are still not addressing the TODO you referenced above, as we should try to keep the scope of this PR focused to the upgrades' logic.